### PR TITLE
Update %load_node docs to make requirements more explicit

### PR DESCRIPTION
**Pull request by noklam**
## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: docs/source/notebooks_and_ipython/kedro_and_notebooks.md [hunk 1]
```diff
@@ -226,15 +226,22 @@ For more details, run `%reload_kedro?`.
 ``` {note}
 This is still an experimental feature and is currently only available for Jupyter Notebook (>7.0), Jupyter Lab, IPython, and VSCode Notebook. If you encounter unexpected behaviour or would like to suggest feature enhancements, add it under [this github issue](https://github.com/kedro-org/kedro/issues/3580).
 ```
+You can load the contents of a node in your project into a series of cells using the `%load_node` line magic. To use `%load_node`, you need to fulfill two requirements:
+- The node need to have a name
```
Comment (astrojuanlu) [inline docs/source/notebooks_and_ipython/kedro_and_notebooks.md:230]:
```suggestion
- The node needs to have a name
```

File: docs/source/notebooks_and_ipython/kedro_and_notebooks.md [hunk 2]
```diff
@@ -226,15 +226,22 @@ For more details, run `%reload_kedro?`.
...
+- The node's inputs need to be persisted
 
-You can load the contents of a node in your project into a series of cells using the `%load_node` line magic.
+You can find [examples of creating node with name](../nodes_and_pipelines/nodes.md#how-to-create-a-node). By default, Kedro save data in memory. To persist the data, you can [register dataset in `DataCatalog`](../tutorial/create_a_pipeline.md#preprocessed-data-registration).
```
Comment (astrojuanlu) [inline docs/source/notebooks_and_ipython/kedro_and_notebooks.md:233]:
```suggestion
You can find [examples of creating nodes with names](../nodes_and_pipelines/nodes.md#how-to-create-a-node). By default, Kedro saves data in memory. To persist the data, you need to [declare the dataset in the Data Catalog](../tutorial/create_a_pipeline.md#preprocessed-data-registration).
```

File: docs/source/notebooks_and_ipython/kedro_and_notebooks.md [hunk 3]
```diff
@@ -226,15 +226,22 @@ For more details, run `%reload_kedro?`.
...
+
+``` {note}
+Node name need to be unique within the pipeline. With the absence of user defined name, Kedro generates one using a combination of the function name, inputs and outputs.
```
Comment (astrojuanlu) [inline docs/source/notebooks_and_ipython/kedro_and_notebooks.md:236]:
```suggestion
The node name needs to be unique within the pipeline. In the absence of a user defined name, Kedro generates one using a combination of the function name, inputs and outputs.
```

within the pipeline or within the project? 🤔

Comment (noklam) [reply docs/source/notebooks_and_ipython/kedro_and_notebooks.md:236]: Technically it's a `Pipeline` object, so in theory you can have the same name in two different Pipeline, definitely a bad practice though.

File: docs/source/notebooks_and_ipython/kedro_and_notebooks.md [hunk 4]
```diff
@@ -226,15 +226,22 @@ For more details, run `%reload_kedro?`.
 ``` {note}
 This is still an experimental feature and is currently only available for Jupyter Notebook (>7.0), Jupyter Lab, IPython, and VSCode Notebook. If you encounter unexpected behaviour or would like to suggest feature enhancements, add it under [this github issue](https://github.com/kedro-org/kedro/issues/3580).
 ```
+You can load the contents of a node in your project into a series of cells using the `%load_node` line magic. To use `%load_node`, you need to fulfill two requirements:
```
Comment (merelcht) [inline docs/source/notebooks_and_ipython/kedro_and_notebooks.md:229]:
```suggestion
You can load the contents of a node in your project into a series of cells using the `%load_node` line magic. To use `%load_node`, the node you want to load needs to fulfil two requirements:
```

File: docs/source/notebooks_and_ipython/kedro_and_notebooks.md [hunk 5]
```diff
@@ -226,15 +226,22 @@ For more details, run `%reload_kedro?`.
...
+- The node needs to have a name
+- The node's inputs need to be persisted
 
-You can load the contents of a node in your project into a series of cells using the `%load_node` line magic.
+You can find [examples of creating nodes with names](../nodes_and_pipelines/nodes.md#how-to-create-a-node). By default, Kedro saves data in memory. To persist the data, you need to [declare the dataset in the Data Catalog](../tutorial/create_a_pipeline.md#preprocessed-data-registration).
```
Comment (merelcht) [inline docs/source/notebooks_and_ipython/kedro_and_notebooks.md:233]:
```suggestion
The [section about creating nodes with names](../nodes_and_pipelines/nodes.md#how-to-create-a-node) explains how to ensure your node has a name. By default, Kedro saves data in memory. To persist the data, you need to [declare the dataset in the Data Catalog](../tutorial/create_a_pipeline.md#preprocessed-data-registration).
```

**Review by astrojuanlu — APPROVED** [review]
LGTM!

File: kedro/framework/cli/micropkg.py [hunk 6]
```diff
@@ -246,7 +246,7 @@ def _pull_package(  # noqa: PLR0913
         package_name = packages[0]
 
         # Type ignored because of https://github.com/pypa/build/pull/693
-        package_reqs = _get_all_library_reqs(library_meta)  # type: ignore[arg-type]
+        package_reqs = _get_all_library_reqs(library_meta)
```
Comment (merelcht) [inline kedro/framework/cli/micropkg.py:249]: Is this change still needed?

Comment (noklam) [reply kedro/framework/cli/micropkg.py:249]: Good catch, not anymore after Ankita changes

**Comment by noklam** [comment]
@merelcht I added a new link in FAQ and added the mention of `%load_ext kedro.ipython` is run by the custom jupyter kernel.

File: docs/source/notebooks_and_ipython/kedro_and_notebooks.md [hunk 7]
```diff
@@ -41,8 +41,10 @@ The `kedro jupyter notebook` command launches a notebook with a customised kerne
 * `pipelines` (type `dict[str, Pipeline]`): Pipelines defined in your [pipeline registry](../nodes_and_pipelines/run_a_pipeline.md#run-a-pipeline-by-name)
 * `session` (type {py:class}`~kedro.framework.session.session.KedroSession`): [Kedro session](../kedro_project_setup/session.md) that orchestrates a pipeline run
 
+In addtion, it also run `%load_ext kedro.ipython` automatically when you launch the notebook.
```
Comment (merelcht) [inline docs/source/notebooks_and_ipython/kedro_and_notebooks.md:44]:
```suggestion
In addtion, it also runs `%load_ext kedro.ipython` automatically when you launch the notebook.
```

**Review by merelcht — APPROVED** [review]
One minor comment, but otherwise looks good 👍
